portfolio/src/test/java/com/google/sps/GenreRecommenderTest.java

```diff
@@ -0,0 +1,135 @@
+package com.google.sps;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.sps.data.Book;
+import com.google.sps.data.BookReader;
+import com.google.sps.data.GenreRecommender;
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * This is a test class for BookReader.java
+ * It only tests the basic functionality for reading from a CSV file.
+ * It does not check that the results match those from the Books API.
+ */
+@RunWith(JUnit4.class)
+public final class GenreRecommenderTest {
+  private List<Book> bookList;
+
+  // ONE GENRE BOOKS:
+  private static final Book AUTOBIOGRAPHY_BOOK =
+      Book.builder()
+          .title("book about me")
+          .genre(new HashSet<>(Arrays.asList("autobiography")))
+          .addReview("N/A")
+          .build();
+  private static final Book BIOGRAPHY_BOOK = Book.builder()
+                                                 .title("book about him")
+                                                 .genre(new HashSet<>(Arrays.asList("biography")))
+                                                 .addReview("N/A")
+                                                 .build();
+  private static final Book CRIME_BOOK = Book.builder()
```

Comment:
opt: can do in separate PR.

It might be helpful to create a static helper function in this test class to create books instance (to avoid redundant code, and improve readability):
````
private static final Book BIOGRAPHY_BOOK = createBook("book about him", "biography");

private static Book createBook(String title, String ...genres) {
  Book.builder()
    .title(title)
    .genre(new HashSet<>(Arrays.asList(genres)))
    .addReview("N/A")
    .build();
}
```

By convention we define the private methods in class after the test cases.